### PR TITLE
Create eEth Pricing Extension

### PR DESCRIPTION
**Pull request by 0xEinCodes**
**## Core changes:**

This PR encompasses the development and unit testing of the eEth Pricing Extensions. 

The main features include:
1. Take the market price of weETH && use the eETH/weETH exchange rate to get price for eETH
    1. This will allow us to use a wrapping adaptor for eETH that allows conversion btw EETH into weETH

**Details of the design:**
1. Query price of weETH from the price router
2. Query the exchange rate between weETH and eETH from the weETH contract 
3. Use those 2 values to get the price of eETH in USD

**References**:

1. [EtherFi Docs](https://etherfi.gitbook.io/etherfi/contracts-and-integrations/deployed-contracts)
2. [Past Pricing Extensions for reference](https://github.com/PeggyJV/cellar-contracts/blob/main/src/modules/price-router/Extensions/Lido/WstEthExtension.sol)
3. [Redstone Price Feeds](https://docs.redstone.finance/docs/smart-contract-devs/price-feeds)

**Lingering TODOs:**

1. confirm that units are [weETH / eETH] for `getRate()`

### CODE REVIEW

File: src/modules/price-router/Extensions/EtherFi/eETHExtension.sol [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    /**
+     * @notice Attempted to use this extension to price something other than eETH.
+     */
+    error eEthExtension__ASSET_NOT_EETH();
+
+    /**
+     * @notice Ethereum mainnet eETH.
+     */
+    address public eETH = 0x35fA164735182de50811E8e2E824cFb9B6118ac2;
```
Comment (crispymangoes) [inline src/modules/price-router/Extensions/EtherFi/eETHExtension.sol:30]: Make these constants or immutable

File: src/modules/price-router/Extensions/EtherFi/eETHExtension.sol [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    /**
+     * @notice Ethereum mainnet weETH.
+     */
+    address public weETH = 0xCd5fE23C85820F7B72D0926FC9b05b43E359b7ee;
```
Comment (crispymangoes) [inline src/modules/price-router/Extensions/EtherFi/eETHExtension.sol:35]: Make these constants or immutable

File: src/modules/price-router/Extensions/EtherFi/eETHExtension.sol (hunk 1 above)
Comment (crispymangoes) [reply src/modules/price-router/Extensions/EtherFi/eETHExtension.sol:30]: lol why did I not make the wsteth ones constant

**Review by crispymangoes — APPROVED** [review]
Make the one small change and we are gucci :) nice work!

File: src/modules/price-router/Extensions/EtherFi/eETHExtension.sol (hunk 1 above)
Comment (crispymangoes) [reply src/modules/price-router/Extensions/EtherFi/eETHExtension.sol:30]: Also make these ERC20s

Comment (0xEinCodes) [reply src/modules/price-router/Extensions/EtherFi/eETHExtension.sol:30]: Changes have been made :)

File: src/modules/price-router/Extensions/EtherFi/eETHExtension.sol (hunk 2 above)
Comment (0xEinCodes) [reply src/modules/price-router/Extensions/EtherFi/eETHExtension.sol:35]: Done, also made it an ERC20

**Pull request merged**